src/main/scala/coreplex/Coreplex.scala

```diff
@@ -65,21 +80,82 @@ abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConf
 }
 
 class DefaultCoreplex(tp: Parameters, tc: CoreplexConfig) extends Coreplex()(tp, tc) {
+  class TileWrapper(tileBuilder: (Bool, Parameters) => Tile, _clock: Clock, _reset: Bool)
+      (implicit p: Parameters) extends Module(Some(_clock), Some(_reset)) {
+
+    val wrappedTile = tileBuilder(reset, p)
+    val io = new Bundle {
+      val tile = wrappedTile.io.cloneType
+      val ext_clock = Clock(INPUT)
+      val ext_reset = Bool(INPUT)
+    }
+
+    io.tile.uncached <> wrappedTile.io.uncached.map(AsyncClientUncachedTileLinkTo(io.ext_clock, io.ext_reset, _))
+    io.tile.cached <> wrappedTile.io.cached.map(AsyncClientTileLinkTo(io.ext_clock, io.ext_reset, _))
+    wrappedTile.io.interrupts := LevelFrom(io.ext_clock, io.tile.interrupts)
```

Comment:
Okay, so for the PRCI, if you fence before exiting the interrupt routine so that you know the update of the timecmp register has been acknowledged, you can be sure that the interrupt has settled.

If you want to be 100% sure of this, you could have an acknowledgement system in which the interrupt gets passed into the tile domain, registered, and then passed back out to the PLIC/PRCI. Then the PLIC/PRCI could check the returned value to make sure it has updated before sending the acknowledgement.